astroquery/alma/core.py

```diff
@@ -236,7 +233,73 @@ def _get_dataarchive_url(self):
 
     def stage_data(self, uids):
         """
-        Stage ALMA data
+        Obtain table of ALMA files
+
+        Parameters
+        ----------
+        uids : list or str
+            A list of valid UIDs or a single UID.
+            UIDs should have the form: 'uid://A002/X391d0b/X7b'
+
+        Returns
+        -------
+        data_file_table : Table
+            A table containing 3 columns: the UID, the file URL (for future
+            downloading), and the file size
+        """
+
+        dataarchive_url = self._get_dataarchive_url()
+
+        # allow for the uid to be specified as single entry
+        if isinstance(uids, str):
+            uids = [uids]
+
+        tables = []
+        for uu in uids:
+            uid = clean_uid(uu)
+            req = self._request('GET', '{dataarchive_url}/rh/data/expand/{uid}'
+                                .format(dataarchive_url=dataarchive_url,
+                                        uid=uid),
+                                cache=False)
+            req.raise_for_status()
+            try:
+                jdata = req.json()
+            except json.JSONDecodeError:
+                if 'Central Authentication Service' in req.text:
+                    # this indicates a wrong server is being used;
+                    # the "pre-feb2020" stager will be phased out
+                    # when the new services are deployed
+                    return self.stage_data_prefeb2020(uids)
+                else:
+                    raise
+            if jdata['type'] != 'PROJECT':
+                log.error(f"Skipped uid {uu} because it is not a project and"
+                          "lacks the appropriate metadata; it is a "
+                          f"{jdata['type']}")
+                continue
+            table = uid_json_to_table(jdata)
+            table['sizeInBytes'].unit = u.B
+            table.rename_column('sizeInBytes', 'size')
+            table.add_column(Column(data=['{dataarchive_url}/dataPortal/{name}'
+                                          .format(dataarchive_url=dataarchive_url,
+                                                  name=name)
+                                          for name in table['name']],
+                                    name='URL'))
+            tables.append(table)
+
+        if len(tables) == 0:
+            raise ValueError("No valid UIDs supplied.")
+
+        table = table_vstack(tables)
+
+        return table
+
+    def stage_data_prefeb2020(self, uids):
+        """
+        Stage ALMA data - old server style
+
+        NOTE: this method will be removed when a new ALMA service is deployed
+        in March 2020
```

Comment:
will we have a follow-up PR to do the cleanup?

Reply:
eh, maybe, let's see if they actually make the upstream change first =)

Reply:
ok. we need a reminder issue or, just remember to clean this up, in ~6 months time or similar.